File: erlB_int.c
#include <stdio.h>
#include <math.h>

float Es( int s, float a );

main()
{
  int    s;
  float a, es;

  printf("Erlang B Formula (positive integer server)\n");

  printf("s="); scanf("%d", &s);
  printf("a="); scanf("%g", &a);

  es = Es(s, a);

  printf("s=%d\ta=%g\tEs(a)=%g\n", s, a, es);

}

float Es( int s, float a )
{
  int    i;
  float es=1.0;
  for(i=1; i<=s; i++)
    es = a * es / (i+a*es);

  return(es);

}

File: erlB_int.go
package main

import "fmt"

func main() {
	var s int
	var a, es float64

	fmt.Println("Erlang B Formula (positive integer server)")
	fmt.Printf("s=")
	fmt.Scanf("%d", &s)
	fmt.Printf("a=")
	fmt.Scanf("%f", &a)

	es = Es(s, a)

	fmt.Printf("s=%d\ta=%f\tEs(a)=%f\n", s, a, es)
}

func Es(s int, a float64) float64 {
	es := 1.0
	for i := 1; i <= s; i++ {
		es = a * es / (float64(i) + a*es)
	}
	return es
}
